src/main/webapp/js/index.js

```diff
@@ -16,7 +16,16 @@
 /** Fetches receipts from the server and adds them to the DOM. */
 async function searchReceipts() {
   const categories = document.getElementById('category-input').value;
-  const response = await fetch(`/search-receipts?categories=${categories}`);
+  const dateRange = document.getElementById('date-range-input').textContent;
+  const store = document.getElementById('store-name-input').value;
+  const minPrice = document.getElementById('min-price-input').value;
+  const maxPrice = document.getElementById('max-price-input').value;
+  const dateTimeFormat = new Intl.DateTimeFormat();
+  const timeZoneId = dateTimeFormat.resolvedOptions().timeZone;
+
+  const response = await fetch(`/search-receipts?categories=
+    ${categories}&store=${store}&min=${minPrice}&max=
+    ${maxPrice}&dateRange=${dateRange}&timeZoneId=${timeZoneId}`);
```

Comment:
I would consider using [URLSearchParams](https://developers.google.com/web/updates/2016/01/urlsearchparams) to build the parameter list or use encodeURIComponent to make sure the user input is properly encoded (e.g., contacting the string above won't handle if the input includes a space like "burger king").